avpipe.go

```diff
@@ -1446,6 +1454,19 @@ func Probe(params *XcParams) (*ProbeInfo, error) {
 		probeInfo.StreamInfo[i].FieldOrder = AVFieldOrderNames[AVFieldOrder(probeArray[i].field_order)]
 		probeInfo.StreamInfo[i].Profile = int(probeArray[i].profile)
 		probeInfo.StreamInfo[i].Level = int(probeArray[i].level)
+		probeInfo.StreamInfo[i].SideData.DisplayMatrix.Rotation = float64(probeArray[i].side_data.display_matrix.rotation)
+
+		dict := (*C.AVDictionary)(probeArray[i].tags)
+		dictCount := (int)(C.dict_count(dict))
+		if dictCount > 0 {
+			probeInfo.StreamInfo[i].Tags = make(map[string]string, dictCount)
+			for c := 0; c < dictCount; c++ {
+				k := C.GoString((*C.char)(C.dict_get_key(dict, (C.int)(c))))
+				v := C.GoString((*C.char)(C.dict_get_value(dict, (C.int)(c))))
+				probeInfo.StreamInfo[i].Tags[k] = v
+			}
+			C.dict_free(dict)
```

Comment:
Can we use directly av_dict_free() here?

C.av_dict_free(&dict)

then we don't need local dict functions.

Reply:
I made it work directly with the built-in av_dict_get() iterator